### DISCUSSION TITLE
RibsPositioning using EtaXsiPoint

### DISCUSSION THREAD

**Discussion by GodertdeZoetenGKNFokker** [DLR-SL/CPACS#744] [Q&A] [answered]
When using the ribsPositioning option startEtaXsiPoint and endEtaXsiPoint, I was wondering what should be done when these points are not on the ribStart or ribEnd curve?
Should the rib just pass through this defined point and be trimmed at the LE/TE or a spar?

Also: what should be done with the locations of the ribs between the start and the end?

Finally: I noticed that in the CPACS 3.0 documentation ribsPositioning does not have a ribReference, but in CPACS 2.3.1 and CPACS 3.0 documentation it does have a ribReference. Also, my example CPACS 3.0 files does have a ribReference. Hence my question: maybe ribReference was accidentally omitted in the CPACS 3.0 documentation?

Kind regards,
Godert

**Comment by MarAlder** [comment, marked as answer]
Hi Godert,

to your last comment: yes, the `ribReference` was accidentally omitted in CPACS 3.0. See [this commit](https://github.com/DLR-SL/CPACS/commit/1551466d6b6942329ecdd541f4fe5688425edb5a). However, TiGL 3 needs this parameter. It has been added in [CPACS 3.1](https://www.cpacs.de/documentation/CPACS_3_1_0_Docs/html/893c8088-24d2-8047-ed69-1d4cf9d883ff.htm).

`RibStart` and `RibEnd` limit the shape extrusion and `startEtaXsiPoint` and `endEtaXsiPoint` are control points defining the beginning and ending of the rib set and thus must not be on the start- or end curve. (@rainman110 or @joergbrech can you confirm this?). 

We should update this figure, but I guess it shows that start and end point must not be on the ribStart or ribEndCurve:
![grafik](https://user-images.githubusercontent.com/43143741/99828724-9ebae980-2b5b-11eb-9f6f-a24292fc6945.png)
